### PR TITLE
Remove AOT

### PR DESCRIPTION
**Pull request by RobinMalfait**
The start of something beautiful. Let's remove AOT mode!

### CODE REVIEW

**Comment by TrySound** [comment]
Does it mean we don't need large bundles inside dist anymore?

**Comment by adamwathan** [comment]
@TrySound Yeah we are removing the CDN builds from this package for v3.0, whatever we make available CDN-wise we'll put in a separate package so we stop wasting everyone's bandwidth 👀

**Comment by RobinMalfait** [comment]
For people interested, this basically has a 10x reduction in size...
| Before | After |
| -- | -- |
| 34.6 MB | 3.7 MB |

image

File: src/cli.js [hunk 1]
```diff
@@ -420,39 +418,21 @@ async function build() {
     let resolvedConfig = resolveConfigInternal(config)
 
     if (args['--purge']) {
-      resolvedConfig.purge = {
-        enabled: true,
-        content: args['--purge'].split(/(?<!{[^}]+),/),
+      log.warn(['The `--purge` flag has been deprecated.', 'Please use `--content` instead.'])
+      if (!args['--content']) {
+        args['--content'] = ['--purge']
```
Comment (binaryoverload) [inline src/cli.js:423]: This should probably be `args['--content'] = args['--purge']`

The current version will just assign an array with `--purge` as an element

Comment (binaryoverload) [reply src/cli.js:423]: Whoops, just realised I commented on a merged branch 🤦‍♂️

Comment (RobinMalfait) [reply src/cli.js:423]: No worries, and thanks for notifying! I've fixed it here: https://github.com/tailwindlabs/tailwindcss/pull/5646

**Comment by lightyen** [comment]
How can I get all available classname from tailwind.config after this change?

**Comment by RobinMalfait** [comment]
@lightyen you cannot get *all* the available classes because there will be millions of those. We do have a way to get all the **base** classes, so that means something like `bg-red-500` without all variants but the API is a bit complex right now and you do require an internal `context` to access these. 

However, I'm more interested in your use case, can you talk more about that and maybe open an issue/discussion so that we can talk about the use case and find solutions?

**Comment by lightyen** [comment]
@RobinMalfait I just need all variants and utilities(or components) for list, if there is a way, it is good!

**Comment by RobinMalfait** [comment]
@lightyen can you open a discussion, then we can move the conversation to its dedicated spot. Thanks! https://github.com/tailwindlabs/tailwindcss/discussions/new

**Pull request merged**